### PR TITLE
[actions] Support Semgrep by Github Actions

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### Why I did it
[Semgrep](https://github.com/returntocorp/semgrep) is a static analysis tool to find security vulnerabilities.
When opening a PR or commtting to PR, Semgrep performs a diff-aware scanning, which scans changed files in PRs.
When merging PR, Semgrep performs a full scan on master branch and report all findings.

Ref: - [Supported Language](https://semgrep.dev/docs/supported-languages/#language-maturity) - [Semgrep Rules](https://registry.semgrep.dev/rule)
#### How I did it
Integrate Semgrep into this repository by committing a job configuration file
#### How to verify it
PR: https://github.com/maipbui/sonic-buildimage/pull/2
Master branch full scan findings: [Master branch findings results](https://github.com/maipbui/sonic-buildimage/actions/runs/3160181876/jobs/5144332404)
PR https://github.com/maipbui/sonic-buildimage/pull/2 scan findings: [Pull request findings results](https://github.com/maipbui/sonic-buildimage/actions/runs/3160193505/jobs/5144357859)

### CODE REVIEW

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Semgrep
+
+on:
+  pull_request: {}
+  push:
+    branches: ["master"]
```
Comment (qiluo-msft) [inline .github/workflows/semgrep.yml:6]: Is it possible to specify `master` and all release branches `[0-9]*` ?

File: .github/workflows/semgrep.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  semgrep:
+    name: Semgrep
+    runs-on: ubuntu-latest
```
Comment (qiluo-msft) [inline .github/workflows/semgrep.yml:11]: Is it possible to run on `debian-latest`?

File: .github/workflows/semgrep.yml (hunk 1 above)
Comment (maipbui) [reply .github/workflows/semgrep.yml]: Updated. This should be backported to older branches.

File: .github/workflows/semgrep.yml (hunk 2 above)
Comment (maipbui) [reply .github/workflows/semgrep.yml:14]: Github-hosted runners do not include debian-latest.

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**